java/dataset/src/main/java/org/apache/arrow/dataset/jni/JniWrapper.java

```diff
@@ -76,6 +76,22 @@ private JniWrapper() {
    */
   public native long createScanner(long datasetId, String[] columns, long batchSize, long memoryPool);
 
+  /**
+   * Create Scanner from a Dataset and get the native pointer of the Dataset.
+   * @param datasetId the native pointer of the arrow::dataset::Dataset instance.
+   * @param columns desired column names.
+   *                Columns not in this list will not be emitted when performing scan operation. Null equals
+   *                to "all columns".
+   * @param filter the filter to apply on the scan
```

Comment:
nit: usage of period at the end of a comment. Certain parameters do have it but some parameters don't.